R/output_combined_table.R

```diff
@@ -0,0 +1,135 @@
+        data_collectors = ifelse(
+          !is.na(ORCID),
+          paste0(data_collectors, " <ORCID:", ORCID),
+          data_collectors),
+        data_collectors = ifelse(
+          is.na(ORCID),
+          paste0(data_collectors, " <affiliation:", affiliation),
+          paste0(data_collectors, ";affiliation:", affiliation)),
+        data_collectors = ifelse(
+          !is.na(additional_role),
+          paste0(data_collectors, ";additional_role:", additional_role, ">"),
+          paste0(data_collectors, ">"))
+      ) %>%
+      dplyr::select(-dplyr::all_of(c("last_name", "given_name", "ORCID", "affiliation", "additional_role"))) %>%
+      dplyr::group_by(dataset_id) %>%
+      dplyr::mutate(data_collectors = paste0(data_collectors, collapse = "; ")) %>%
+      dplyr::ungroup() %>%
+      dplyr::distinct()
+
+  contexts_tmp <-
+    database$contexts %>%
+      dplyr::mutate(
+        context_property = stringr::str_replace_all(context_property, "=", "-"),
+        value = stringr::str_replace_all(value, "=", "-"),
+        description = stringr::str_replace_all(description, "=", "-"),
+        context_property = stringr::str_replace_all(context_property, ";", ","),
+        value = stringr::str_replace_all(value, ";", ","),
+        description = stringr::str_replace_all(description, "=", "-"),
+        value = ifelse(
+          is.na(description),
+          paste0(context_property, ":", value),
+          paste0(context_property, ":", value, " <", description, ">"))
+      ) %>%
+    dplyr::select(-dplyr::all_of(c("description", "context_property", "category"))) %>%
+    tidyr::separate_longer_delim(link_vals, ", ") %>%
+    distinct()
+
+  reformat_contexts <- function(contexts_table, context_id) {
+    context_category <- gsub("_id", "_properties", context_id, fixed = TRUE)
+    out <- contexts_table %>%
+      dplyr::filter(link_id == context_id) %>%
+      dplyr::select(-link_id) %>%
+      dplyr::distinct(dataset_id, link_vals, .keep_all = TRUE)
+
+    names(out)[which(names(out) == "value")] <- context_category
+    names(out)[which(names(out) == "link_vals")] <- context_id
+    out
+  }
+
+  join_contexts <- function(data, contexts_tmp) {
+    data %>%
+      dplyr::left_join(
+      by = c("dataset_id", "treatment_context_id"),
+      reformat_contexts(contexts_tmp, "treatment_context_id")
+    ) %>%
+    dplyr::left_join(
+      by = c("dataset_id", "plot_context_id"),
+      reformat_contexts(contexts_tmp, "plot_context_id")
+    ) %>%
+    dplyr::left_join(
+      by = c("dataset_id", "entity_context_id"),
+      reformat_contexts(contexts_tmp, "entity_context_id")
+    ) %>%
+    dplyr::left_join(
+      by = c("dataset_id", "temporal_context_id"),
+      reformat_contexts(contexts_tmp, "temporal_context_id")
+    ) %>%
+    dplyr::left_join(
+      by = c("dataset_id", "method_context_id"),
+      reformat_contexts(contexts_tmp, "method_context_id")
+    )
+  }
+
+  combined_table <-
```

Comment:
I'm wondering if the ID columns that help join to other relational tables could be removed from the combined table?